vllm/entrypoints/openai/serving_chat.py

```diff
@@ -1074,6 +1074,14 @@ async def chat_completion_stream_generator(
 
                     # if the model is finished generating
                     else:
+                        # check for error finish reason and abort streaming
+                        if error_data := self._handle_streaming_error_finish_reason(
+                            output.finish_reason, request_id
+                        ):
```

Comment:
do you want to add the same check in speech_to_text.py (audio endpoints) for completeness?